src/GraphQL/Validation/ValidationResult.cs

```diff
@@ -50,5 +67,14 @@ public sealed class SuccessfullyValidatedResult : IValidationResult
         /// Returns an empty list of execution errors.
         /// </summary>
         public ExecutionErrors Errors => EmptyExecutionErrors.Instance;
+
+        /// <inheritdoc/>
+        public Variables? Variables => null;
```

Comment:
I doubt about these nulls. This class is public and now it can be used ONLY if `variables == null && argumentValues == null && directiveValues == null`. Existing code may break silently. I would make `SuccessfullyValidatedResult` internal.

Reply:
> I would make `SuccessfullyValidatedResult` internal.

Agree

Reply:
I mean the name of the class does not correspond to the contents. It is some sort of `SuccessfullyValidatedResultWithoutAnyVariablesAndArgumentsAndDirectives`.